File: prime.cpp
#include <stdio.h>
#include <math.h> 
#include <iostream>


bool * generatePrimes(int n){
    int i,j,k;
    int nSqrt = sqrt(n);
    bool array[nSqrt] = { 1 };

    for(int i=0; i<nSqrt; i++) {
        if(array[i]) {
            j = pow(i,i);
            k=0;
            while(j < n) {
                array[j] = 0;
                k+=1;
                j+=k*i;
            }
        }
    }
    return array;
}

int main() {
    int n;
    bool *listPrime;
    
    cin >> n;

    listPrime = generatePrimes(n);

    return 0;
}

File: prime.py
import math

def generatePrimes(n):
    nSqrt = int(math.sqrt(n))
    array = [True for i in range(nSqrt)]

    for i in range(1,nSqrt):
        if array[i] == True:
            j = i**2
            k=0
            while j < n:
                print(j)
                array[j] = False
                k+=1
                j+=k*i

    return array

n = int(input())
primeList = generatePrimes(n)
